go/cmd/betabot/main.go

```diff
@@ -57,64 +62,326 @@ func betabot() error {
 		qrterminal.GenerateHalfBlock(res.HTMLURL, qrterminal.L, os.Stdout)
 	}
 
+	var wg sync.WaitGroup
+	// request contact in flag
+	{
+		if *contact != "" {
+			splits := strings.Split(*contact, ",")
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				for _, v := range splits {
+					parseRet, err := messengerClient.ParseDeepLink(ctx, &bertymessenger.ParseDeepLink_Request{Link: v})
+					if err != nil {
+						log.Printf("contact was not valid")
+						return
+					}
+					{
+						om, err := proto.Marshal(&bertymessenger.ContactMetadata{DisplayName: *displayName})
+						if err != nil {
+							cancel()
+							return
+						}
+						_, err = messengerClient.SendContactRequest(ctx, &bertymessenger.SendContactRequest_Request{BertyID: parseRet.BertyID, OwnMetadata: om})
+						if err != nil {
+							cancel()
+							return
+						}
+					}
+				}
+			}()
+		}
+	}
+
 	// event loop
 	{
 		s, err := messengerClient.EventStream(ctx, &bertymessenger.EventStream_Request{})
 		if err != nil {
+			log.Printf("Failed to listen EventStream %v", err)
 			return err
 		}
-
-		handlerMutex := sync.Mutex{}
+		wg.Add(1)
 		go func() {
+			defer wg.Done()
+
 			for {
 				gme, err := s.Recv()
 				if err != nil {
+					cancel()
 					log.Printf("stream error: %v", err)
+					return
 				}
-				handlerMutex.Lock()
-				err = handleEvent(ctx, messengerClient, gme)
-				handlerMutex.Unlock()
-				if err != nil {
-					log.Printf("event handler error: %v", err)
-				}
+
+				wg.Add(1)
+				go func() {
+					handleEvent(ctx, messengerClient, gme)
+					wg.Done()
```

Comment:
```suggestion
					defer wg.Done()
					handleEvent(ctx, messengerClient, gme)
```
Safer in case of panic (defer are executed when a panic occurs)